spring-cloud-netflix-eureka-server/src/main/java/org/springframework/cloud/netflix/eureka/server/EurekaServerAutoConfiguration.java

```diff
@@ -275,12 +285,32 @@ public EurekaServerConfig eurekaServerConfig(EurekaClientConfig clientConfig) {
 	static class RefreshablePeerEurekaNodes extends PeerEurekaNodes
 			implements ApplicationListener<EnvironmentChangeEvent> {
 
+		private ReplicationClientAdditionalFilters replicationClientAdditionalFilters;
+
 		RefreshablePeerEurekaNodes(final PeerAwareInstanceRegistry registry,
 				final EurekaServerConfig serverConfig,
-				final EurekaClientConfig clientConfig, final ServerCodecs serverCodecs,
-				final ApplicationInfoManager applicationInfoManager) {
-			super(registry, serverConfig, clientConfig, serverCodecs,
-					applicationInfoManager);
+				final EurekaClientConfig clientConfig,
+				final ServerCodecs serverCodecs,
+				final ApplicationInfoManager applicationInfoManager,
+				final ReplicationClientAdditionalFilters replicationClientAdditionalFilters) {
+			super(registry, serverConfig, clientConfig, serverCodecs, applicationInfoManager);
+			this.replicationClientAdditionalFilters = replicationClientAdditionalFilters;
+		}
+
+		@Override
+		protected PeerEurekaNode createPeerEurekaNode(String peerEurekaNodeUrl) {
+			JerseyReplicationClient replicationClient = JerseyReplicationClient
```

Comment:
Does this work when you do not use Jersey and use RestTemplate?

Reply:
My understanding is that peer replication calls are all handled in Netflix eureka-core with those statically created JerseyReplicationClient for each peer node. (https://github.com/Netflix/eureka/blob/8325086bb6012841ab85e0288d3ceec743b79a76/eureka-core/src/main/java/com/netflix/eureka/cluster/PeerEurekaNodes.java#L197) 
Is there a way to use RestTemplate instead of Jersey?

Reply:
You can configure the Eureka client to not use jersey, see https://cloud.spring.io/spring-cloud-static/Greenwich.SR1/single/spring-cloud.html#_eurekaclient_without_jersey

I am unsure if this will effect peep to peer replication as well, but I would like to confirm if it does or doesnt

Reply:
There is, but it looks like it might need to be extended for peer replication. The rest template stuff is mostly for eureka clients. I'm not too worried about it here. If there is ever a need we can certainly open an issue.